### PR TITLE
Modifed Maya Camera

### PR DESCRIPTION
**Pull request by Przemog1**

### CODE REVIEW

**Comment by Przemog1** [comment]
> wtf.. that will only work on MSVC ?

fixed

File: source/Irrlicht/CSceneNodeAnimatorCameraModifiedMaya.h [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+			virtual void setDistance(float distance);
+
+			//! This animator will receive events when attached to the active camera
+			virtual bool isEventReceiverEnabled() const
+			{
+				return true;
+			}
+
+			//! Enable zooming with right mouse button
+			void enableRMBZoom()
```
Comment (devshgraphicsprogramming) [inline source/Irrlicht/CSceneNodeAnimatorCameraModifiedMaya.h:75]: we don't need any of the non-virtual functions

File: source/Irrlicht/CSceneNodeAnimatorCameraModifiedMaya.h [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+		public:
+			//! Constructor
+			CSceneNodeAnimatorCameraModifiedMaya(gui::ICursorControl* cursor, float rotateSpeed = -1500.f,
+				float zoomSpeed = 200.f, float translationSpeed = 1500.f, float distance = 70.f);
+
+			//! Animates the scene node, currently only works on cameras
+			virtual void animateNode(IDummyTransformationSceneNode* node, uint32_t timeMs);
+
+			//! Event receiver
+			virtual bool OnEvent(const SEvent& event);
```
Comment (devshgraphicsprogramming) [inline source/Irrlicht/CSceneNodeAnimatorCameraModifiedMaya.h:42]: whenever you use `virtual` use the `override` keyword in the base classes

File: source/Irrlicht/CSceneManager.h [hunk 3]
```diff
@@ -113,6 +113,11 @@ namespace scene
 			float translationSpeed=1500.f, int32_t id=-1, float distance=70.f,
 			bool makeActive=true);
 
+		virtual ICameraSceneNode* addCameraSceneNodeModifiedMaya(IDummyTransformationSceneNode* parent = 0,
+			float rotateSpeed = -1500.f, float zoomSpeed = 200.f,
+			float translationSpeed = 1500.f, int32_t id = -1, float distance = 70.f,
+			bool makeActive = true);
```
Comment (devshgraphicsprogramming) [inline source/Irrlicht/CSceneManager.h:119]: need the parameter about RMB zoom here

**Comment by devshgraphicsprogramming** [comment]
can you make a cleaner PR that doesn't have 49 modified files when done against the `brdf_explorer` branch?

(as well as fix my comments above)

**Pull request closed without merging**